src/toast/Toast.tsx

```diff
@@ -0,0 +1,170 @@
+    type?: MessageType;
+    icon?: string;
+    text: string;
+    onClose: Function;
+    style?: any;
+    className?: string;
+};
+
+type ToastState = {
+    showToast: boolean;
+};
+
+export class Toast extends React.PureComponent<ToastProps> {
+    private divRef: HTMLDivElement | null = null;
+    private _isMounted = false;
+
+    state: ToastState = {
+        showToast: this.props.showToast !== undefined ? this.props.showToast : false,
+    };
+
+    // Default value for props if no value is given
+    static defaultProps = {
+        showToast: false,
+        type: MessageType.SUCCESS,
+        position: Direction.TOP_RIGHT,
+        duration: DEFAULT_HIDE_TIME,
+        icon: IconType.SUCCESS,
+    };
+
+    componentWillUpdate(nextProps: ToastProps, nextState: ToastState) {
+        if (this.showToast(nextProps, nextState)) {
+            this.createElement();
+
+            if (nextProps.showToast === true && nextProps.showToast !== nextState.showToast) {
+                let duration = nextProps.duration === undefined ? DEFAULT_HIDE_TIME : nextProps.duration;
+                //set the timer for the toast
+                this.setTimer(duration);
+            }
+        } else {
+            this.cleanup();
+        }
+    }
+
+    componentWillMount() {
+        this.createElement();
+    }
+
+    createElement() {
+        if (this.divRef === null) {
+            const el = document.createElement("div");
+            document.body.appendChild(el);
+            this.divRef = el;
+        }
+        document.body.classList.add(ClassNames.NO_SCROLLING);
+    }
+
+    componentDidMount() {
+        this._isMounted = true;
+    }
+
+    setTimer(hideTime: number) {
+        setTimeout(() => {
+            this.close();
+        }, hideTime);
+    }
+
+    cleanup() {
+        if (this.divRef !== null) {
+            document.body.removeChild(this.divRef);
+            this.divRef = null;
+        }
+        document.body.classList.remove(ClassNames.NO_SCROLLING);
+    }
+
+    componentWillUnmount() {
+        this.close();
+        this._isMounted = false;
+        this.cleanup();
+    }
+
+    buildToast(): React.ReactElement {
+        const {text, icon, position, type, duration, showToast, style, className} = this.props;
+
+        let styled = {
+            ...style,
+            padding: "20px 20px",
+            zIndex: 2000,
+            position: "fixed",
+        };
+
+        //create styles based on the position given
+        let place = (position || Direction.TOP_RIGHT).includes(Directions.BOTTOM) ? Directions.BOTTOM : Directions.TOP;
+        styled[place] = 0;
+        place = (position || Direction.TOP_RIGHT).includes(Directions.RIGHT) ? Directions.RIGHT : Directions.LEFT;
+        styled[place] = 0;
+
+        return (
+            <div className={classNames(["alert", `alert-${type}`, className])} style={styled}>
+                <span style={{paddingRight: "5px"}}>{icon && <Icon shape={icon} />}</span>
```

Comment:
Check if user has passed icon then only render this span tag

Reply:
Addressed comment